android/java/org/chromium/chrome/browser/firstrun/WelcomeOnboardingActivity.java

```diff
@@ -0,0 +1,337 @@
+            }
+        });
+
+        mBtnNegative.setOnClickListener(view -> {
+            if (mCurrentStep == 2) {
+                CustomTabActivity.showInfoPage(this, P3A_URL);
+            } else {
+                nextOnboardingStep();
+            }
+        });
+    }
+
+    private void startTimer(int delayMillis) {
+        new Handler().postDelayed(this::nextOnboardingStep, delayMillis);
+    }
+
+    private void nextOnboardingStep() {
+        mCurrentStep++;
+
+        if (mCurrentStep == 0) {
+            int margin = mIsTablet ? 100 : 0;
+            setLeafAnimation(mVLeafAlignTop, mIvLeafTop, 1f, margin, true);
+            setLeafAnimation(mVLeafAlignBottom, mIvLeafBottom, 1f, margin, false);
+            setFadeInAnimation(mTvWelcome, 200);
+            mIvBrave.animate().scaleX(0.8f).scaleY(0.8f).setDuration(1000);
+            new Handler().postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    mTvWelcome.animate()
+                            .translationYBy(-dpToPx(WelcomeOnboardingActivity.this, 20))
+                            .setDuration(3000)
+                            .start();
+                }
+            }, 200);
+
+            startTimer(3000);
+
+        } else if (mCurrentStep == 1) {
+            int margin = mIsTablet ? 200 : 30;
+            setLeafAnimation(mVLeafAlignTop, mIvLeafTop, 1.3f, margin, true);
+            setLeafAnimation(mVLeafAlignBottom, mIvLeafBottom, 1.3f, margin, false);
+
+            if (BraveSetDefaultBrowserUtils.isBraveSetAsDefaultBrowser(this)) {
+                mBtnPositive.setText(getResources().getString(R.string.continue_text));
+                mBtnNegative.setVisibility(View.GONE);
+            }
+            mTvWelcome.setVisibility(View.GONE);
+            mLayoutCard.setVisibility(View.VISIBLE);
+            mIvArrowDown.setVisibility(View.VISIBLE);
+
+        } else if (mCurrentStep == 2) {
+            int margin = mIsTablet ? 250 : 60;
+            setLeafAnimation(mVLeafAlignTop, mIvLeafTop, 1.5f, margin, true);
+            setLeafAnimation(mVLeafAlignBottom, mIvLeafBottom, 1.5f, margin, false);
+
+            mLayoutCard.setVisibility(View.GONE);
+            mTvDefault.setVisibility(View.GONE);
+            mIvArrowDown.setVisibility(View.GONE);
+
+            mTvCard.setText(getResources().getString(R.string.p3a_title));
+            mBtnPositive.setText(getResources().getString(R.string.continue_text));
+            mBtnNegative.setText(getResources().getString(R.string.learn_more_onboarding));
+
+            if (PackageUtils.isFirstInstall(this)
+                    && (getPackageName().equals(BraveConstants.BRAVE_NIGHTLY_PACKAGE_NAME)
+                            || getPackageName().equals(BraveConstants.BRAVE_BETA_PACKAGE_NAME))
+                    && !OnboardingPrefManager.getInstance().isP3aCrashReportingMessageShown()) {
+                mCheckboxCrash.setChecked(true);
```

Comment:
this is not correct. The value should always default to `isUsageAndCrashReportingPermittedByUser` as specified https://github.com/brave/reviews/issues/1132#issuecomment-1335818255 and in our privacy policy

Reply:
actually sorry, this predates that, but this code has not changed so it is not incorrect